packages/forms/frontend/sirius-components-forms/src/representations/FormRepresentation.tsx

```diff
@@ -76,120 +51,70 @@ const useFormRepresentationStyles = makeStyles()((theme) => ({
-  const { classes } = useFormRepresentationStyles();
-  const [{ value, context }, dispatch] = useMachine<FormRepresentationContext, FormRepresentationEvent>(
-    formRepresentationMachine,
-    {
-      context: {
-        formId: representationId,
-      },
-    }
-  );
-  const { toast, formRepresentation } = value as SchemaValue;
-  const { id, formId, form, message } = context;
   const [state, setState] = useState<FormRepresentationState>({
     payload: null,
+    form: null,
   });
 
-  /**
-   * Displays an other form if the selection indicates that we should display another properties view.
-   */
+  const { payload } = useFormSubscription(editingContextId, representationId);
   useEffect(() => {
-    if (formId !== representationId) {
-      const switchFormEvent: SwitchFormEvent = { type: 'SWITCH_FORM', formId: representationId };
-      dispatch(switchFormEvent);
+    if (isFormRefreshedEventPayload(payload)) {
+      setState((prevState) => ({ ...prevState, form: payload.form }));
     }
-  }, [representationId, formId, dispatch]);
+  }, [payload]);
 
-  const { error } = useSubscription<GQLFormEventSubscription>(formEventSubscription, {
-    variables: {
-      input: {
-        id,
-        editingContextId,
-        formId: representationId,
-      },
-    },
-    fetchPolicy: 'no-cache',
-    onData: ({ data }) => {
-      if (data.data) {
-        const { formEvent } = data.data;
-        setState((prevState) => ({ ...prevState, payload: formEvent }));
-      }
-      const handleDataEvent: HandleSubscriptionResultEvent = {
-        type: 'HANDLE_SUBSCRIPTION_RESULT',
-        result: data,
-      };
-      dispatch(handleDataEvent);
-    },
-    onComplete: () => {
-      const completeEvent: HandleCompleteEvent = { type: 'HANDLE_COMPLETE' };
-      dispatch(completeEvent);
-    },
-  });
-
-  useEffect(() => {
-    if (error) {
-      const { message } = error;
-      const showToastEvent: ShowToastEvent = { type: 'SHOW_TOAST', message };
-      dispatch(showToastEvent);
-    }
-  }, [error, dispatch]);
+  const { classes } = useFormRepresentationStyles();
 
-  let content: JSX.Element | null = null;
-  if (formRepresentation === 'ready') {
-    if (form.pages.length > 1) {
-      content = <Form editingContextId={editingContextId} form={form} readOnly={readOnly} />;
-    } else if (form.pages.length === 1) {
-      let selectedPageToolbar = null;
-      if (form.pages[0].toolbarActions?.length > 0) {
-        selectedPageToolbar = (
-          <div className={classes.toolbar}>
-            {form.pages[0].toolbarActions.map((toolbarAction) => (
-              <div className={classes.toolbarAction} key={toolbarAction.id}>
-                <ToolbarAction
-                  editingContextId={editingContextId}
-                  formId={form.id}
-                  readOnly={readOnly}
-                  widget={toolbarAction}
-                />
-              </div>
-            ))}
+  const renderedForm = useMemo(() => {
```

Comment:
Please stop with the `useMemo` `memo`, `useCallback`etc unless we have a performance issue identified that needs to be fixed (i.e. a Github issue). Premature optimization does not help and React 19 is supposed to provide that for free so we will have to remove this kind of code quite soon.

Reply:
Seemed worth it to me there, but I also took the wrong habit to use these too much.
I'll remove it there.